### ISSUE TITLE
Modify the code to use torch.nn.TripletMarginLoss(margin = margin, p = 2)

### ISSUE THREAD

**Issue by jpainam** [layumi/Person-reID-triplet-loss#6]
Hi, I found out that you implemented a triplet loss instead of using `torch.nn.TripletMarginLoss(margin = margin, p = 2)` or `nn.SoftMarginLoss()` if no margin.
https://github.com/layumi/Person-reID-triplet-loss/blob/15c80480100bf7294f663c3e07a0c47d022333b4/train_new.py#L256

1) Is there any improvement in your implementation?
2) Which are the `neg_feat` and `pos_feat` to consider if i want to use `torch.nn.TripletMarginLoss(margin=1.0)` 
Thank you

**Comment by layumi** [comment]
Hi @jpainam 
1. I have tried several loss terms, and find that the loss sum works well.  Yes, I also felt weird....

2. You may consider `pf_hard` and `nf_hard`. 
https://github.com/layumi/Person-reID-triplet-loss/blob/master/train_new.py#L246-L247

**Comment by jpainam** [comment]
Thank you.
